shuffle.py

```diff
@@ -57,6 +58,13 @@ def log_4(n):
         i += 1
     return i
 
+def normalize(array_in, limit=16384):
```

Comment:
The limit of 128 *128 is due to the output image size being hardcoded to 128* 128. Otherwise, the existing hilbert png code should, I think, pad the input list until it has a number of elements equal to the next power of 4. Maybe instead of truncating list at 128 * 128, could instead dynamically change the image size to be 128x128 or 256x256... etc.